### PR TITLE
Handle RequestRedirect errors properly

### PR DESCRIPTION
**Pull request by barakalon**
A proposed fix for https://github.com/plangrid/flask-rebar/issues/24

### CODE REVIEW

**Comment by barakalon** [comment]
Looks like this is incompatible with Flask 0.10.*... working on it...

File: flask_rebar/rebar.py [hunk 1]
```diff
@@ -47,6 +50,14 @@
 ])
 
 
+# To catch redirection exceptions, app.errorhandler expects 301 in versions
+# below 0.11.0, but the exception itself in version greater than 0.11.0.
+if version.parse(flask_version) < version.parse('0.11.0'):
+    redirect_error = 301
```
[the remark this answers is not in the document]
Comment (barakalon) [reply flask_rebar/rebar.py]: Ah, sure.

File: tests/test_rebar.py [hunk 2]
```diff
@@ -508,3 +509,21 @@ def force_500():
...
+
+        app = create_rebar_app(rebar)
+
+        app.debug = False
+        resp = app.test_client().get(path='/with_trailing_slash')
+        self.assertEqual(resp.status_code, 301)
+        self.assertEqual(resp.content_type, 'application/json')
+        self.assertTrue(resp.headers['Location'].endswith('/with_trailing_slash/'))
+
+        app.debug = True
```
[the remark this answers is not in the document]
Comment (barakalon) [reply tests/test_rebar.py]: I think flask-rebar should reliably return JSON when not in debug mode, which is why the behavior is a little different than flask's built in behavior.

It _does_ set the `Location` header properly, so browsers will still automatically redirect.

File: flask_rebar/rebar.py [hunk 3]
```diff
@@ -15,11 +15,14 @@
 from collections import namedtuple
 from copy import copy
 from functools import wraps
+from packaging import version
```
Comment (brockhaywood) [inline flask_rebar/rebar.py]: For some reason, i need to install `packaging` specifically using pip3 in python 3.5

Comment (barakalon) [reply flask_rebar/rebar.py]: Ah, crap. I should've used another way of doing this comparison

Comment (barakalon) [reply flask_rebar/rebar.py]: Let me change this real quick

Comment (barakalon) [reply flask_rebar/rebar.py]: https://github.com/plangrid/flask-rebar/pull/29

**Pull request merged**